packages/frames.js/src/core/middleware/concurrentMiddleware.ts

```diff
@@ -20,17 +20,28 @@ export function parallelizeMiddleware<
 
   return async (context, next) => {
     let ctx = context;
+    const newContexts: any[] = [];
 
     await Promise.all(
       middlewares.map((middleware) =>
         // @ts-expect-error this is not 100% correct because next should return a valid result, but since parallel middleware cannot call next middleware in chain, it's fine
         middleware(context, (newCtx) => {
-          ctx = { ...ctx, ...newCtx };
+          if (newCtx) {
+            newContexts.push(newCtx);
+          }
         })
       )
     );
 
+    let finalCtx = ctx;
+
+    for (const newCtx of newContexts) {
+      if (newCtx) {
+        finalCtx = { ...finalCtx, ...newCtx };
+      }
+    }
```

Comment:
sorry, nitpick - this null check is already done above, is there a reason to check again?

Reply:
you are right, removed it.